File: Python/Mahasiswa.py
class Mahasiswa:
    __nim = ""
    __nama = ""
    __prodi = ""
    __fakultas = ""

    def __init__(self, nim = "", nama = "", prodi = "", fakultas = ""):
        self.__nim = nim
        self.__nama = nama
        self.__prodi = prodi
        self.__fakultas = fakultas

    # Getter #
    #NIM
    def getNim(self):
        return self.__nim
    #Nama
    def getNama(self):
        return self.__nama
    #Prodi
    def getProdi(self):
        return self.__prodi
    #Fakultas
    def getFakultas(self):
        return self.__fakultas
    
    # Setter #
    #NIM
    def setNim(self, nim):
        self.__nim = nim
    #Nama
    def setNama(self, nama):
        self.__nama = nama
    #Prodi
    def setProdi(self, prodi):
        self.__prodi = prodi
    #Fakultas
    def setFakultas(self, fakultas):
        self.__fakultas = fakultas
    #SetAll use for Edits
    def setAll(self, nama, prodi, fakultas):
        self.__nama = nama
        self.__prodi = prodi
        self.__fakultas = fakultas

    # Method #
    def carNim(self, nim):
        if self.__nim == nim:
            return 1
        else:
            return 0

File: C++/Mahasiswa.cpp
#include <iostream>
#include <string>

using namespace std;

class Mahasiswa{
    private:
        string nim;
        string nama;
        string prodi;
        string fakultas;

    public:
        /*Constructors*/
        Mahasiswa(){
            this->nim = "";
            this->nama = "";
            this->prodi = "";
            this->fakultas = "";
        }
        /*Constructors*/
        Mahasiswa(string nim, string nama, string prodi, string fakultas){
            this->nim = nim;
            this->nama = nama;
            this->prodi = prodi;
            this->fakultas = fakultas;
        }

        /*Setter*/
        //NIM
        void setNim(string nim){
            this->nim = nim;
        }
        //Nama
        void setNama(string nama){
            this->nama = nama;
        }
        //Prodi
        void setProdi(string prodi){
            this->prodi = prodi;
        }
        //Fakultas
        void setFakultas(string fakultas){
            this->fakultas = fakultas;
        }

        /*Getter*/
        //NIM
        string getNim(){
            return this->nim;
        }
        //Nama
        string getNama(){
            return this->nama;
        }
        //Prodi
        string getProdi(){
            return this->prodi;
        }
        //Fakultas
        string getFakultas(){
            return this->fakultas;
        }

        /*Method*/
        void setAll(string nama, string prodi, string fakultas){
            this->nama = nama;
            this->prodi = prodi;
            this->fakultas = fakultas;
        }

        int carNim(string nim){
            if(this->nim.compare(nim) == 0){
                return 1;
            }else{
                return 0;
            }
        }

};